vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/ComboBox.java

```diff
@@ -207,8 +241,15 @@ public void initialize() {
     // provided.
     private String lastFilter;
 
+    /*
+     * Holds the item count listeners count.
+     * If the value > 0, then the client side filter will be sent to server
+     * forcibly, in order to let the items size updated and item count change
+     * event triggered (if, of course, the size has changed).
```

Comment:
Isn't client side filtering used only with in-memory data ? It seems just very odd to me that the whole filtering mode would change when the only reason why this is necessary is that the server side needs to know the amount of items.

Reply:
> Isn't client side filtering used only with in-memory data ?

Yes, if `ListDataProvider` or generic `InMemoryDataProvider` is used and items count < page size, then client filter works without requests to server. Basically this is how it works before data view changes. I've added it for `InMemoryDataProvider` and forced server requests in case of listeners.

Reply:
> whole filtering mode would change when the only reason why this is necessary is that the server side needs to know the amount of items.

Mayhaps, but IMO more odd DX will be expect item count change triggering, but getting nothing because of client filter mode.

Well, when I explored it, it took some time for me to figure out why I didn't get those terrible event...oh, because of only 7 items :)

Reply:
This code has been removed. Another issue is that the client filter change should not trigger item count change event on the server. But this will be a separate ticket.